### ISSUE TITLE
4.9.39 fails with undefined reference to "current_thread_info" in binfmt_elf.c

### ISSUE THREAD

**Issue by karesmakro** [minipli/linux-unofficial_grsec#6]
Kernel 4.9.38 was fine, but I'm not able to compile the latest one, which fails on:
```
fs/built-in.o: In Function `load_elf_binary':
binfmt_elf.c:(.text+0x9a710): undefined reference to `current_thread_info'
```
Thank you very much for your port!

**Comment by minipli** [comment]
Thanks for the report! I messed it up, sorry.

I assume you're building a 32 bit build with SEGMEXEC disabled? ... Anyways, will have a fix tonight.

**Comment by karesmakro** [comment]
Yes SEGMEXEC was disabled!

**Comment by ghost** [comment]
I have the same issue on x86-64 build. I use config from offcial debian packages https://packages.debian.org/sid/linux-image-4.9.0-2-grsec-amd64

[config-4.9.0-2-grsec-amd64.txt](https://github.com/minipli/linux-unofficial_grsec/files/1167450/config-4.9.0-2-grsec-amd64.txt)

**Comment by theLOICofFRANCE** [comment]
The problem must come from this:
https://patchwork.kernel.org/patch/9807291/

**Comment by minipli** [comment]
Should be fixed by commit 7779e98dd5f3 ("fs: binfmt_elf.c - include <linux/sched.h> first"), i.e. v4.9.39-unofficial_grsec-20170723002624.diff.

**Comment by minipli** [comment]
Just for the record: It wasn't related to SEGMEXEC nor 32 bit builds but 64 bit builds with enabled CONFIG_COMPAT.
